docs/src/UsingQAT/TargetingQIR.md

```diff
@@ -0,0 +1,248 @@
+has support for this.
+
+### Target QIS mapping - Rules used to transform instruction sequences in the QIR
+
+This adaptor transforms the QIR into a QIR that is targeted with respect to the
+quantum instruction set. The adaptor provides a number of different
+transformations including mapping dynamic qubit/result allocation into static
+qubit/result allocation and optimize comparisons against constant quantum
+results `Zero` and `One`:
+
+| Name                              | Description                                                                                           | Default value |
+| --------------------------------- | ----------------------------------------------------------------------------------------------------- | ------------- |
+| optimize-result-one               | Maps branching based on quantum measurements compared to one to base adaptor type measurement.        | true          |
+| optimize-result-zero              | Maps branching based on quantum measurements compared to zero to base adaptor type measurement.       | true          |
+| optimize-result-comparison        | Maps branching based on quantum measurements compared to each other to base adaptor type measurement. | true          |
+| use-static-qubit-array-allocation | Maps allocation of qubit arrays to static array allocation.                                           | true          |
+| use-static-qubit-allocation       | Maps qubit allocation to static allocation.                                                           | true          |
+| use-static-result-allocation      | Maps result allocation to static allocation.                                                          | true          |
+| disable-reference-counting        | Disables reference counting by instruction removal.                                                   | true          |
+| disable-alias-counting            | Disables alias counting by instruction removal.                                                       | true          |
+| disable-string-support            | Disables string support by instruction removal.                                                       | true          |
+| disable-record-output-support     | Disables record output support by instruction removal.                                                | true          |
+| reuse-qubits                      | Use to define whether or not to reuse qubits.                                                         | false         |
+| reuse-results                     | Use to define whether or not to reuse results.                                                        | false         |
+| entry-point-attr                  | Specifies the attribute indicating the entry point.                                                   | EntryPoint    |
+
+It is worth noting that the mapping of dynamic qubits to static qubits is
+limited in scope and done as a best-effort. The reason for this limitation is
```

Comment:
It would be helpful to know what the limitations are exactly. Is this described in more detail in the code?